### ISSUE TITLE
Comparing between field names

### ISSUE THREAD

**Issue by xescuder** [deitch/searchjs#48]
I have an array with elements like:

{
    start:5
    end: 6    
}

There's any way to do a search in array (matchArray) to find all objects where {start:{gte:end}} or {start:{gte:end-1}} ?

I'm interested in doing that sort of things, and I think your library is great, but I need something like this.
Thanks!

**Comment by xescuder** [comment]
Yes, exactly. That's what I need. I need also to check for example if (start gte end*0.75).

**Comment by deitch** [comment]
No support for it now, but no reason it couldn't be extended to do so. The comparisons take place entirely in [this function](https://github.com/deitch/searchjs/blob/master/src/searchjs.js#L17). It would need to be extended to be able to handle `field` being not just an absolute value, but perhaps an object that could contain relative fields. 

Would also need a way to change the syntax in the query.

Go for a PR?
